modules/atomic-widgets/styles/atomic-styles-manager.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Elementor\Modules\AtomicWidgets\Styles;
+
+use Elementor\Core\Breakpoints\Breakpoint;
+use Elementor\Core\Utils\Collection;
+use Elementor\Modules\AtomicWidgets\Cache;
+use Elementor\Plugin;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Atomic_Styles_Manager {
+	private static ?self $instance = null;
+
+	private array $registered_styles_by_key = [];
+
+	private array $post_ids = [];
+
+	const DEFAULT_BREAKPOINT = 'desktop';
+
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	public function register_hooks() {
+		add_action( 'elementor/frontend/after_enqueue_post_styles', fn() => $this->enqueue_styles() );
+		add_action( 'elementor/post/render', function( $post_id ) {
+			$this->post_ids[] = $post_id;
+		} );
+	}
+
+	public function register( string $key, callable $get_style_defs ) {
+		$this->registered_styles_by_key[ $key ] = $get_style_defs;
+	}
+
+	private function enqueue_styles() {
```

Comment:
I think it will be more clear If you will prepare the data before in some structure like
```ts
{
	'desktop-base-style': () => {},
	'mobile-base-style': () => {},
	'desktop-elements-200': () => {},
	...
}
```

Then you will have only one cache and not multiple.